core/src/main/scala/org/apache/spark/FutureAction.scala

```diff
@@ -229,28 +183,15 @@ class ComplexFutureAction[T] extends FutureAction[T] {
       processPartition: Iterator[T] => U,
       partitions: Seq[Int],
       resultHandler: (Int, U) => Unit,
-      resultFunc: => R) {
+      resultFunc: => R) : FutureAction[R] = synchronized {
```

Comment:
because you return a future now the java doc is out of date (it doesn't actually run a job)